### PR TITLE
DS-432 link deprecations

### PR DESCRIPTION
**Pull request by remydenton**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-432

## Summary

Removes deprecated props from link component

## Details

- Removes four previously deprecated props -- `href`, `isHeadline`, `onClick`, and `onClickTarget` -- from the link component
- Removes all remaining usages of these props in patter lab
- Fixes a bug in microjourneys editing caused by usage of the deprecated `isHeadline` prop.

## How to test

- Confirm that none of the deprecated props -- `href`, `isHeadline`, `onClick`, and `onClickTarget` -- work in the link component.

- Go to /pattern-lab/?p=experiments-micro-journeys
- Confirm that there are no errors when clicking the `edit` button (currently, there are errors on boltdesignsystem.com)

## Release notes

- Four previously deprecated props -- `href`, `isHeadline`, `onClick`, and `onClickTarget` -- have been removed from the link component.

### CODE REVIEW

**Comment by colbytcook** [comment]
@remydenton I did test the microjouney "isHeadline" (changing line 765 in the setup-bolt.js) to "is_headline" and it seemed to work fine (no console errors and was able to edit and save)

**Review by colbytcook — APPROVED** [review]

**Comment by remydenton** [comment]
Thanks @colbytcook for the testing.  I actually already fixed `isHeadline` in this PR (by changing it to `is_headline`, just as you suggest).  You should be able to test following the testing instructions once the deployment is fixed.

**Review by cjwhitedev — APPROVED** [review]
everything looks right to me

**Review by danielamorse — APPROVED** [review]

File: packages/components/bolt-link/src/link.twig [hunk 1]
```diff
@@ -6,26 +6,6 @@
   {{ validate_data_schema(schema, _self) | raw }}
 {% endif %}
 
-{# DEPRECATED.  Use the property `url` instead of `href`. #}
```
Comment (danielamorse) [inline packages/components/bolt-link/src/link.twig:9]: @cjwhitedev love how easy it is to delete when we deprecate props this way.

**Pull request merged**